mongoexport/mongoexport.go

```diff
@@ -378,6 +367,27 @@ func (exp *MongoExport) exportInternal(out io.Writer) (int64, error) {
 // of documents successfully exported, and a non-nil error if something went wrong
 // during the export operation.
 func (exp *MongoExport) Export(out io.Writer) (int64, error) {
+	session, err := exp.SessionProvider.GetSession()
+	if err != nil {
+		return 0, err
+	}
+
+	coll := session.Database(exp.ToolOptions.Namespace.DB).Collection(exp.ToolOptions.Namespace.Collection)
+	collInfo, err := db.GetCollectionInfo(coll)
+	if err != nil {
+		return 0, err
+	}
+
+	// If the collection doesn't exist, GetCollectionInfo will return nil
+	if collInfo == nil {
```

Comment:
Before the port, we weren't checking if the collection exists up front. I added this logic because I didn't want to add nil checks every time we call GetCollectionInfo in the code.

Reply:
I considered it but decided against it to minimize diff noise. I'm open to doing this in another commit on this ticket or on a different ticket entirely if we think it's a good idea.

Reply:
Added a cached version of CollectionInfo in the latest commit.